revup/toolkit.py

```diff
@@ -56,5 +57,11 @@ async def main(args: argparse.Namespace, git_ctx: git.Git) -> int:
     elif args.toolkit_cmd == "closest-branch":
         await git_ctx.verify_branch_or_commit(args.branch[0])
         logging.info(await git_ctx.get_best_base_branch(args.branch[0], allow_self=args.allow_self))
+    elif args.toolkit_cmd == "list-topics":
+        topics = TopicStack(git_ctx, args.base_branch, args.relative_branch)
+        await topics.populate_topics()
+        for topic in topics.topics.values():
+            topic_commits = " ".join(commit.commit_id for commit in topic.original_commits)
```

Comment:
is the commit id specifically useful here? i'd think users would prefer having commit title instead

i'd recommend the following:

list-topics just prints topics newline separated
with --commit-id, lists commit ids newline separated as well. so there'd be 2 newlines between different topics
with --title, lists commit titles newline separated similar to above

that should in theory work with what you want and is more flexible. newline is so we don't run over the edge of the screen with topics that have a lot of commits

Reply:
> is the commit id specifically useful here? i'd think users would prefer having commit title instead

Yes the commit id is much more useful for scripts that parse the output.

> list-topics just prints topics newline separated
> with --commit-id, lists commit ids newline separated as well. so there'd be 2 newlines between different topics
> with --title, lists commit titles newline separated similar to above

Done. Might be worth considering doing something like "<topic name> <number of commits>" followed by the list of commits instead of seperating topics by the double-newline, which I think is marginally simpler to parse from a script  (and matches what git does in several places).